File: cards-be/Entities/Deck.cs
namespace API.Entites;

public class Deck
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string? Description { get; set; }
    public bool IsPrivate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // navigation properties
    public int UserId { get; set; }
    public User User { get; set; }
    public int? ClassId { get; set; }
    public Class? Class { get; set; }

    public List<Card> Cards { get; set; } = [];
}

File: cards-fe/src/app/models/deck.ts
import { Flashcard } from "./card";

export interface Deck {
  id: number;
  title: string;
  description: string;
  isPrivate: boolean;
  createdAt: string;
  updatedAt: string;
  classId: number;
  username: string;
  cardsCount: number;
}

export interface DeckWithCards {
  id: number;
  title: string;
  description: string;
  isPrivate: boolean;
  createdAt: string;
  updatedAt: string;
  classId: number;
  username: string;
  cards: Flashcard[];
}
